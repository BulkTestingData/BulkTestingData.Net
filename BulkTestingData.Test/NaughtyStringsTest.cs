using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace BulkTestingData.Test
{
    [TestClass]
    [UseReporter(typeof(DiffReporter), typeof(ClipboardReporter))]
    public class NaughtyStringsTest
    {
        [TestMethod]
        public void TestAll()
        {
            Approvals.VerifyAll("length", NaughtyStrings.AllExceptNull(), s => "{0} => {1}".FormatWith(s, s.Length));
        }

        [TestMethod]
        public void TestAllAndNull()
        {
            Approvals.VerifyAll("length", NaughtyStrings.All().Take(5), s => "{0} => {1}".FormatWith(s, String.IsNullOrEmpty(s)));
        }

    }
}
