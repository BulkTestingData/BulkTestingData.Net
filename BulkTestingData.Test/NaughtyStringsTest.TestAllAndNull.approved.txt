length

 => True
 => True
undefined => False
undef => False
null => False
